src/backend/marsha/core/views.py

```diff
@@ -76,31 +120,32 @@ def get_context_data(self):
         except ImproperlyConfigured:
             pass
 
-        if video is not None:
+        if resource is not None:
             # Create a short-lived JWT token for the video
             jwt_token = AccessToken()
             jwt_token.payload.update(
                 {
                     "session_id": str(uuid.uuid4()),
-                    "video_id": str(video.id),
                     "context_id": lti.context_id,
+                    "resource_id": str(resource.id),
                     "roles": lti.roles,
                     "course": lti.get_course_info(),
                     "locale": locale,
                     "read_only": lti.is_student
-                    or video.playlist.lti_id != lti.context_id,
+                    or resource.playlist.lti_id != lti.context_id,
                 }
             )
             try:
                 jwt_token.payload["user_id"] = lti.user_id
             except AttributeError:
                 pass
 
+            self._enrich_jwt_token(jwt_token, resource)
+
             context["jwt_token"] = str(jwt_token)
 
-        context["video_data"] = json.dumps(
-            VideoSerializer(video).data if video else None
-        )
+        context["resource_data"] = self._get_resource_data(resource)
```

Comment:
With my proposal above this would become:
```
context["video_data"] = json.dumps(
    self.serializer_class(resource).data if resource else None
)
```